Build errors - CentOS

Tried to build but running into several problems.  

- cmake 3 is required, installed and being at least partially detected but the executable is called cmake3, not cmake.   Thus aom_test and build fail.    Unfortunately I don't know much about rust to dig into why but a grep on the build directory shows references to it however building fails.   It appears the fgrep for cmake in README.md overrides everything?

- Rust dependency is almost 300M (can cmake be used directly?) -- ultimately I believe causing the final build error which I cannot decipher:


```
./build.sh 
File ‘nyan.y4m’ already there; not retrieving.

   Compiling rav1e v0.1.0 (file:///tmp/build/rav1e)
error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/transform.rs:90:46
   |
90 |     for (raw, clamped) in buffer[c..].iter().step_by(4).zip(temp_in.iter_mut())
   |                                              ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/transform.rs:96:50
   |
96 |       temp_out.iter().zip(output[c..].iter_mut().step_by(stride).take(4)) {
   |                                                  ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/me.rs:35:29
   |
35 |       for y in (y_lo..y_hi).step_by(1) {
   |                             ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/me.rs:36:31
   |
36 |         for x in (x_lo..x_hi).step_by(1) {
   |                               ^^^^^^^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `rav1e`.

To learn more, run the command again with --verbose.

```

Build.sh does not of course accept --verbose however manually adding it results in:

```
ile ‘nyan.y4m’ already there; not retrieving.

       Fresh cc v1.0.22
       Fresh pkg-config v0.3.13
       Fresh unicode-width v0.1.5
       Fresh libc v0.2.43
       Fresh rustc-demangle v0.1.9
       Fresh rand_core v0.2.1
       Fresh cfg-if v0.1.5
       Fresh strsim v0.7.0
       Fresh vec_map v0.8.1
       Fresh ansi_term v0.11.0
       Fresh bitflags v1.0.4
       Fresh y4m v0.3.1
       Fresh bitstream-io v0.6.5
       Fresh cmake v0.1.33
       Fresh textwrap v0.10.0
       Fresh atty v0.2.11
       Fresh rand v0.5.5
       Fresh num-traits v0.2.5
       Fresh clap v2.32.0
       Fresh backtrace-sys v0.1.24
       Fresh backtrace v0.3.9
   Compiling rav1e v0.1.0 (file:///tmp/build/wrk/rav1e)
     Running `rustc --crate-name rav1e src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C codegen-units=1 --cfg 'feature="binaries"' --cfg 'feature="clap"' --cfg 'feature="default"' --cfg 'feature="y4m"' -C metadata=93c5d00589a931bd -C extra-filename=-93c5d00589a931bd --out-dir /tmp/build/wrk/rav1e/target/release/deps -L dependency=/tmp/build/wrk/rav1e/target/release/deps --extern clap=/tmp/build/wrk/rav1e/target/release/deps/libclap-67aa0e1eeb14bfc2.rlib --extern y4m=/tmp/build/wrk/rav1e/target/release/deps/liby4m-1945691963f3ff33.rlib --extern num_traits=/tmp/build/wrk/rav1e/target/release/deps/libnum_traits-d0b1dd0f4bf80312.rlib --extern backtrace=/tmp/build/wrk/rav1e/target/release/deps/libbacktrace-54a555b0f65b75ff.rlib --extern rand=/tmp/build/wrk/rav1e/target/release/deps/librand-d217ea676dd8b85f.rlib --extern bitstream_io=/tmp/build/wrk/rav1e/target/release/deps/libbitstream_io-88e3f0dafbfba2ed.rlib --extern libc=/tmp/build/wrk/rav1e/target/release/deps/liblibc-bd99db9ca5c592bd.rlib -L native=/tmp/build/wrk/rav1e/target/release/build/rav1e-1566397806c53ed0/out/lib -l static=aom -l m -l pthread -L native=/tmp/build/wrk/rav1e/target/release/build/backtrace-sys-787f90aab90eab76/out`
error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/transform.rs:90:46
   |
90 |     for (raw, clamped) in buffer[c..].iter().step_by(4).zip(temp_in.iter_mut())
   |                                              ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/transform.rs:96:50
   |
96 |       temp_out.iter().zip(output[c..].iter_mut().step_by(stride).take(4)) {
   |                                                  ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/me.rs:35:29
   |
35 |       for y in (y_lo..y_hi).step_by(1) {
   |                             ^^^^^^^

error[E0658]: use of unstable library feature 'iterator_step_by': unstable replacement of Range::step_by (see issue #27741)
  --> src/me.rs:36:31
   |
36 |         for x in (x_lo..x_hi).step_by(1) {
   |                               ^^^^^^^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `rav1e`.

Caused by:
  process didn't exit successfully: `rustc --crate-name rav1e src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C codegen-units=1 --cfg feature="binaries" --cfg feature="clap" --cfg feature="default" --cfg feature="y4m" -C metadata=93c5d00589a931bd -C extra-filename=-93c5d00589a931bd --out-dir /tmp/build/wrk/rav1e/target/release/deps -L dependency=/tmp/build/wrk/rav1e/target/release/deps --extern clap=/tmp/build/wrk/rav1e/target/release/deps/libclap-67aa0e1eeb14bfc2.rlib --extern y4m=/tmp/build/wrk/rav1e/target/release/deps/liby4m-1945691963f3ff33.rlib --extern num_traits=/tmp/build/wrk/rav1e/target/release/deps/libnum_traits-d0b1dd0f4bf80312.rlib --extern backtrace=/tmp/build/wrk/rav1e/target/release/deps/libbacktrace-54a555b0f65b75ff.rlib --extern rand=/tmp/build/wrk/rav1e/target/release/deps/librand-d217ea676dd8b85f.rlib --extern bitstream_io=/tmp/build/wrk/rav1e/target/release/deps/libbitstream_io-88e3f0dafbfba2ed.rlib --extern libc=/tmp/build/wrk/rav1e/target/release/deps/liblibc-bd99db9ca5c592bd.rlib -L native=/tmp/build/wrk/rav1e/target/release/build/rav1e-1566397806c53ed0/out/lib -l static=aom -l m -l pthread -L native=/tmp/build/wrk/rav1e/target/release/build/backtrace-sys-787f90aab90eab76/out` (exit code: 101)
```

```
 $ rpm -qa |egrep '(cargo|rust-)'
rust-std-static-1.27.2-3.el7.x86_64
cargo-1.27.0-3.el7.x86_64
rust-1.27.2-3.el7.x86_64
```

--- a/build.sh
+++ b/build.sh
@@ -106,7 +106,9 @@ cargo run --bin rav1e $BUILD_TYPE -- $SEQ -o $ENC_FILE -s 3 -r $REC_FILE
 ${AOM_TEST}/aomdec $ENC_FILE -o $DEC_FILE
 
 # Input/Output compare
-cmp <(tail -n+2 $DEC_FILE) <(tail -n+2 $REC_FILE)
+tail -n+2 $DEC_FILE > /tmp/dec_file
+tail -n+2 $REC_FILE > /tmp/rec_file
+cmp /tmp/dec_file /tmp/rec_file
 
 # Daala tools support coming soon
 #DAALA_TOOLS="../daala/tools/"

--- a/build.sh
+++ b/build.sh
@@ -47,30 +47,40 @@ ACTUAL_VERSION=$(git submodule status | xargs)
 AOM_TEST="aom_test"
 if [[ "$ACTUAL_VERSION" != "$EXPECTED_VERSION" ]] || [[ ! -f ./${AOM_TEST}/aomdec ]]; then
 
-# Store current version to file
-echo $ACTUAL_VERSION > $GITHASH
+    # Store current version to file
+    echo $ACTUAL_VERSION > $GITHASH
 
-# Update aombuild
-git submodule update --init
+    # Update aombuild
+    git submodule update --init
 
-# Clean project files
-cargo clean
+    # Clean project files
+    cargo clean
 
-# Get configure command from readme
-CONFIGURE_CMD=$(fgrep "cmake ../aom" README.md)
+    # Get configure command from readme
+    CONFIGURE_CMD=$(fgrep "cmake ../aom" README.md)
 
-# Wipe and create aom_test folder
-rm -fR $AOM_TEST
-mkdir -p $AOM_TEST
-pushd $AOM_TEST
+    # Handle cases where the cmake binary is absent or named differently
+    if ! [ -x "$(command -v cmake)" ]; then
+        if ! [ -x "$(command -v cmake3)" ]; then
+            echo "\`cmake\` is required to build libaom. Aborting." >&2
+            exit 1
+        fi
 
-echo CONFIGURE COMMAND
-echo $CONFIGURE_CMD
-eval $CONFIGURE_CMD
+        CONFIGURE_CMD=${CONFIGURE_CMD/cmake/cmake3}
+    fi
 
-# auto detect the number of cores and parallel build
-make -j$(nproc --all)
-popd
+    # Wipe and create aom_test folder
+    rm -fR $AOM_TEST
+    mkdir -p $AOM_TEST
+    pushd $AOM_TEST
+
+    echo CONFIGURE COMMAND
+    echo $CONFIGURE_CMD
+    eval $CONFIGURE_CMD
+
+    # auto detect the number of cores and parallel build
+    make -j$(nproc --all)
+    popd
 
 fi
 

